includes/admin/class-wp-job-manager-promoted-jobs-admin.php

```diff
@@ -114,6 +128,54 @@ public function add_action_notice( $actions_handled ) {
 		return $actions_handled;
 	}
 
+	/**
+	 * Add the host of the Promote Job form to the array of allowed redirect hosts.
+	 *
+	 * @param array $hosts Allowed redirect hosts.
+	 * @return array Updated array of allowed redirect hosts.
+	 */
+	public function add_to_allowed_redirect_hosts( $hosts ) {
+		$hosts[] = wp_parse_url( WP_Job_Manager_Helper_API::get_api_base_url(), PHP_URL_HOST );
+		return $hosts;
+	}
+
+	/**
+	 * Handle the action to promote a job listing, validating as well as redirecting to the form on WPJobManager.com.
+	 *
+	 * @return void
+	 */
+	public function handle_promote_job() {
+		$post_id = absint( $_GET['post_id'] ?? 0 );
+		check_admin_referer( self::PROMOTE_JOB_ACTION . '-' . $post_id );
+		if ( ! $post_id ) {
+			wp_die( esc_html__( 'No job listing ID provided for promotion.', 'wp-job-manager' ), '', [ 'back_link' => true ] );
+		}
+		if ( ! is_ssl() ) {
```

Comment:
The requirement to use SSL is mostly to guarantee that there's no "man-in-the-middle" between WPJMCOM and WPJM.

I think I can definitely use `wp_get_environment_type()` to check if it is a production site to enforce this, but, honestly, I'm not sure. Adding a filter is also an option.  But, again, I'm a bit worried about the possible security implications of allowing the user to do this. :thinking:

Reply:
Ah, good idea about the `wp_get_environment_type()`!

> worried about the possible security implications of allowing the user to do this

In the worst case, I think it just would not work, right? Or do you think of any other implication?

Reply:
I'm mainly worried about the user seeing the filter/option to disable the check and just...going in and disabling the check, you know?

I think I can solve that by adding a check directly on WPJMCOM, instead of adding it here. Maybe it would be better than adding here? (at least, we would have a bigger chance to document the whole issue)

Reply:
Hey there! I decided to remove this check in 247e46c and moved the logic to WPJMCOM here: [Automattic/wpjobmanager.com@`f9e7a0c` (#479)](https://github.com/Automattic/wpjobmanager.com/pull/479/commits/f9e7a0c3dc923c4107375b70eadfab0098fc8d51)